program-cache/test/main.cpp

```
@@ -0,0 +1,72 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * OpenCL is a trademark of Apple Inc. used under license by Khronos.
+ */
+
+/// @file main.cpp
+/// @brief Very simple consumption test for OpenCL.
+///
+/// This program is intended to be run with OPENCL_LAYERS=<path-to-ProgramCacheLayer-library>.
+/// It is only able to check if the successfully loaded layer has apparent problems,
+/// but will not signal if the layer load was unsuccessful.
+
+#include <CL/opencl.hpp>
+
+#include <iostream>
+#include <string>
+
+int main()
+{
+    try
+    {
+        auto context = cl::Context::getDefault();
+
+        const std::string program_source = "kernel void foo(global int* i){ *i = 100; }";
+        const cl::Program program(context, program_source);
+        program.build();
+
+        cl::KernelFunctor<cl::Buffer> kernel(program, "foo");
+        const cl::Buffer output(context, CL_MEM_WRITE_ONLY, sizeof(cl_int));
+
+        kernel(cl::EnqueueArgs(cl::NDRange(1)), output);
+
+        cl_int result{};
+        cl::enqueueReadBuffer(output, true, 0, sizeof(result), &result);
+
+        std::cout << "Result: " << result << std::endl;
+        if (result != 100)
+        {
+            return -1;
+        }
+    } catch (const cl::BuildError& ex)
+    {
+        std::cout << "Error: Could not build OpenCL program\n";
+        for (auto [device, log] : ex.getBuildLog())
+        {
+            const auto device_name = device.getInfo<CL_DEVICE_NAME>();
+            std::cout << device_name << ":\n" << log << "\n";
+        }
+        return -1;
+    } catch (const cl::Error& ex)
+    {
+        std::cout << "An OpenCL error occurred: " << ex.what() << " (" << ex.err() << ")\n";
+        return -1;
```

If it's an OpenCL error, I'd probably return the error code as the exit code.

---

Done as requested.